### ISSUE TITLE
Mercator projection range calculations are incorrect

### ISSUE THREAD

**Issue by ossama-othman** [ossama-othman/MaRC#58]
Calculations in MaRC's Transverse Mercator projection assume that the poles (+/- 90 degrees) may be mapped.  That is incorrect since the distortion in that projection approaches infinity at the poles.  A user-configurable maximum latitude should be supported for the Mercator projection in a way similar to the Polar Stereographic projection does that.

Furthermore, the `xmax` calculation in the `MaRC::Mercator::plot_map()` and `plot_grid()` methods seem to assume that `xmax` is a latitude.  It is not.  The `MaRC::Mercator::mercator_x()` should be used to determine `xmax` at the maximum configured latitude range.  Determine whether or not the code needs to be corrected.

**Comment by ossama-othman** [comment]
Some parts of this issue have been corrected in commit 09455a8.  More fixes to come.

**Comment by ossama-othman** [comment]
Upon closer examination, the value `xmax` is not assumed to be a latitude as originally surmised.  `Xmax` is derived like so:

```
 lines    2 * xmax
------- = --------
samples    2 * pi  (radians)

        lines
xmax = ------- * pi
       samples
```

`Xmax` scales with the number of lines in the map.  The greater the number of lines in the map, the greater maximum latitude.
